test/micro/org/openjdk/bench/vm/compiler/VectorBulkOperationsArray.java

```diff
@@ -0,0 +1,700 @@
+             "80", "81", "82", "83", "84", "85", "86", "87", "88", "89",
+             "90", "91", "92", "93", "94", "95", "96", "97", "98", "99",
+            "100","101","102","103","104","105","106","107","108","109",
+            "110","111","112","113","114","115","116","117","118","119",
+            "120","121","122","123","124","125","126","127","128","129",
+            "130","131","132","133","134","135","136","137","138","139",
+            "140","141","142","143","144","145","146","147","148","149",
+            "150","151","152","153","154","155","156","157","158","159",
+            "160","161","162","163","164","165","166","167","168","169",
+            "170","171","172","173","174","175","176","177","178","179",
+            "180","181","182","183","184","185","186","187","188","189",
+            "190","191","192","193","194","195","196","197","198","199",
+            "200","201","202","203","204","205","206","207","208","209",
+            "210","211","212","213","214","215","216","217","218","219",
+            "220","221","222","223","224","225","226","227","228","229",
+            "230","231","232","233","234","235","236","237","238","239",
+            "240","241","242","243","244","245","246","247","248","249",
+            "250","251","252","253","254","255","256","257","258","259",
+            "260","261","262","263","264","265","266","267","268","269",
+            "270","271","272","273","274","275","276","277","278","279",
+            "280","281","282","283","284","285","286","287","288","289",
+            "290","291","292","293","294","295","296","297","298","299",
+            "300"})
+    public static int NUM_ACCESS_ELEMENTS;
+
+    // Every array has two regions:
+    // - read region
+    // - write region
+    // We should make sure that the region is a multiple of 4k, so that the
+    // 4k-aliasing prevention trick can work.
+    //
+    // It would be ince to set REGION_SIZE statically, but we want to keep it rather small if possible,
+    // to avoid running out of cache. But it might be quite large if NUM_ACCESS_ELEMENTS is large.
+    public static int REGION_SIZE = -1024;
+    public static final int REGION_2_BYTE_OFFSET   = 1024 * 2; // prevent 4k-aliasing
+    public static final int REGION_2_SHORT_OFFSET  = REGION_2_BYTE_OFFSET / 2;
+    public static final int REGION_2_CHAR_OFFSET   = REGION_2_BYTE_OFFSET / 2;
+    public static final int REGION_2_INT_OFFSET    = REGION_2_BYTE_OFFSET / 4;
+    public static final int REGION_2_LONG_OFFSET   = REGION_2_BYTE_OFFSET / 8;
+    public static final int REGION_2_FLOAT_OFFSET  = REGION_2_BYTE_OFFSET / 4;
+    public static final int REGION_2_DOUBLE_OFFSET = REGION_2_BYTE_OFFSET / 8;
+    // For Objects, it could be 4 or 8 bytes. Dividing by 8 gives us something
+    // reasonable for both cases.
+    public static final int REGION_2_OBJECT_OFFSET = REGION_2_BYTE_OFFSET / 8;
+
+    // The arrays with the two regions each
```

Comment:
Is there a reason you don't want to have 2 arrays, one as `dst` and one as `src`?